modules/platforms/dotnet/Apache.Ignite/Table/ITableView.cs

```diff
@@ -31,15 +32,148 @@ public interface ITableView<T>
+        /// <param name="key">A record with key columns set.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains a record with all columns.
+        /// </returns>
+        Task<T?> GetAsync(T key);
+
+        /// <summary>
+        /// Gets multiple records by keys.
+        /// </summary>
+        /// <param name="keys">Collection of records with key columns set.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains matching records with all columns.
+        /// </returns>
+        Task<IList<T>> GetAllAsync(IEnumerable<T> keys);
 
         /// <summary>
         /// Inserts a record into the table if it does not exist or replaces the existing one.
         /// </summary>
-        /// <param name="rec">Record to upsert.</param>
+        /// <param name="record">Record to upsert.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        Task UpsertAsync(T rec);
+        Task UpsertAsync(T record);
+
+        /// <summary>
+        /// Inserts multiple records into the table, replacing existing ones.
+        /// </summary>
+        /// <param name="records">Records to upsert.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task UpsertAllAsync(IEnumerable<T> records);
+
+        /// <summary>
+        /// Inserts a record into the table if it does not exist or replaces the existing one.
+        /// </summary>
+        /// <param name="record">Record to upsert.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains replaced record or null if it did not exist.
+        /// </returns>
+        Task<T?> GetAndUpsertAsync(T record);
+
+        /// <summary>
+        /// Inserts a record into the table if it does not exist.
+        /// </summary>
+        /// <param name="record">Record to insert.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains a value indicating whether the record was inserted. Returns <c>false</c> if a
+        /// record with the same key already exists.
+        /// </returns>
+        Task<bool> InsertAsync(T record);
+
+        /// <summary>
+        /// Inserts multiple records into the table, skipping existing ones.
+        /// </summary>
+        /// <param name="records">Records to insert.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous operation.
+        /// The task result contains skipped records.
+        /// </returns>
+        Task<IList<T>> InsertAllAsync(IEnumerable<T> records);
+
+        /// <summary>
+        /// Replaces a record with the same key columns if it exists.
```

Comment:
```suggestion
        /// Replaces a record with the same key columns only if it exists.
```
Probably we should somehow clarify that nothing happens if the record does not exist.

Reply:
Fixed.